HLTrigger/JetMET/plugins/HLTJetHFCleaner.h

```diff
@@ -0,0 +1,219 @@
+
+      if (applyCornerCut_) {
+        auto const inCorner = (sigmaEtaEta < cornerCutSigmaEtaEta_) and (sigmaPhiPhi < cornerCutSigmaPhiPhi_);
+        passSigmaEtaPhiCut &= !inCorner;
+      }
+      if (!passSigmaEtaPhiCut) {
+        continue;
+      }
+    }
+
+    if (applyStripSizeCut_) {
+      // Cut related to central strip size
+      if (centralEtaStripSize >= centralEtaStripSizeThresh_) {
+        continue;
+      }
+    }
+
+    cleaned_jets->emplace_back(jet);
+  }
+
+  iEvent.put(std::move(cleaned_jets));
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+template <typename JetType>
+void HLTJetHFCleaner<JetType>::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("jets", edm::InputTag("hltAK4PFJetsTightIDCorrected"))->setComment("Input jet collection.");
+  desc.add<edm::InputTag>("mets", edm::InputTag("hltMet"))->setComment("Input MET collection.");
+  desc.add<edm::InputTag>("sigmaEtaEta", edm::InputTag("hltHFJetShowerShape", "sigmaEtaEta"))
+      ->setComment("Input collection which stores the sigmaEtaEta values per jet.");
+  desc.add<edm::InputTag>("sigmaPhiPhi", edm::InputTag("hltHFJetShowerShape", "sigmaPhiPhi"))
+      ->setComment("Input collection which stores the sigmaPhiPhis values per jet.");
+  desc.add<edm::InputTag>("centralEtaStripSize", edm::InputTag("hltHFJetShowerShape", "centralEtaStripSize"))
+      ->setComment("Input collection which stores the central strip size values per jet.");
+  desc.add<double>("jetPtThresh", 100)
+      ->setComment("The minimum pt value for a jet such that the cuts will be applied.");
+  desc.add<double>("dphiJetMetThresh", 2.5)
+      ->setComment(
+          "The minimum value for deltaPhi between jet and MET, such that the cuts will be applied to a given jet.");
+  desc.add<double>("jetEtaMin", 2.9)->setComment("Minimum value of jet |eta| for which the cuts will be applied.");
+  desc.add<double>("jetEtaMax", 5.0)->setComment("Maximum value of jet |eta| for which the cuts will be applied.");
+  desc.add<double>("sigmaEtaPhiDiffThresh", 0.05)
+      ->setComment("Determines the threshold in the following cut: sigmaEtaEta-sigmaPhiPhi < X");
+  desc.add<double>("cornerCutSigmaEtaEta", 0.02)
+      ->setComment(
+          "Corner cut value for sigmaEtaEta. Jets will be cut if both sigmaEtaEta and sigmaPhiPhi are lower than the "
+          "corner value.");
+  desc.add<double>("cornerCutSigmaPhiPhi", 0.02)
+      ->setComment(
+          "Corner cut value for sigmaPhiPhi. Jets will be cut if both sigmaEtaEta and sigmaPhiPhi are lower than the "
+          "corner value.");
+  desc.add<int>("centralEtaStripSizeThresh", 3)
+      ->setComment("Determines the threshold in the following cut: centralEtaStripSize < X");
+  desc.add<bool>("applyCornerCut", true)->setComment("Boolean specifying whether to apply the corner cut.");
```

Comment:
Does it make sense to name this `applySigmaEtaPhiCornerCut` (or similar)? I'm nitpicking, but `applyCornerCut` only applies if `applySigmaEtaPhiCut_` is true, and it'd be good to sort-of see that from its name.

Reply:
Thanks for the comments, I think it makes sense to do that yes, I renamed the parameter in [34a2e2a](https://github.com/cms-sw/cmssw/pull/37539/commits/34a2e2a191571dc91b191212fab00330e984cef7).